specification.md

```diff
@@ -157,406 +157,329 @@ The Serverless Workflow language is composed of:
 }
 ```
 
-We will get back to this function definition in just a bit, but now let's take a look at using 
-an inline expression that sets an input parameter inside an action for example:
+We can use an action data filter to filter only the breads data:
 
 ```json
 {
-"actions": [
-    {
-        "functionRef": {
-            "refName": "confirmApplicant",
-            "parameters": {
-                "applicantName": "${ .applicant.name }"
-            }
-        }
+"actions":[  
+    {  
+       "functionRef": "breadAndPastaTypesFunction",
+       "actionFilter": {
+          "results": "${ {breads: .breads} }"
+       }
     }
-]
-}
-```
-
-In this case our input parameter `applicantName` would be set to "John Doe".
-
-Expressions can also be used to select and manipulate state data, this is in particularly useful for 
-state data filters. 
-For example let's use another inline expression:
-
-```json
-{
-   "stateDataFilter": {
-       "dataOutputPath": "${ .applicant | {applicant: .name, contactInfo: { email: .email, phone: .phoneNumbers }} }"
-   }
+ ]
 }
 ```
 
-This would set the data output of the particular state to:
+The `results` will filter the action results which would then be:
 
 ```json
 {
-  "applicant": "John Doe",
-  "contactInfo": {
-    "email": "johndoe@something.com",
-    "phone": [
-      {
-        "type": "iPhone",
-        "number": "0123-4567-8888"
-      },
-      {
-        "type": "home",
-        "number": "0123-4567-8910"
-      }
-    ]
-  }
+  "breads": [
+    "baguette",
+    "brioche",
+    "rye"
+  ]
 }
 ```
 
-[Switch state](#Switch-State) [conditions](#switch-state-dataconditions) require for expressions to be resolved to a boolean value (true / false). 
-
-We can now get back to our previously defined "IsAdultApplicant" expression function and reference it:
+Now let's take a look at a similar example (same expected action results) and assume our current state data is:
 
 ```json
 {
-  "dataConditions": [ {
-    "condition": "${ fn:IsAdultApplicant }",
-    "transition": "StartApplication"
-  }]
+  "itemsToBuyAtStore": [
+  ]
 }
 ```
 
-As previously mentioned, expressions are evaluated against certain subsets of data. For example 
-the `parameters` param of the [functionRef definition](#FunctionRef-Definition) can evaluate expressions 
-only against the data that is available to the [action](#Action-Definition) it belongs to.
-One thing to note here are the top-level [workflow definition](#Workflow-Definition) parameters. Expressions defined
-in them can only be evaluated against the initial [workflow data input](#Workflow-Data-Input).
-
-For example let's say that we have a workflow data input of:
+and have the following action definition:
 
 ```json
 {
-   "inputVersion" : "1.0.0"
+"actions":[  
+    {  
+       "functionRef": "breadAndPastaTypesFunction",
+       "actionFilter": {
+          "results": "${ [ .breads[0], .pasta[1] ] }",
+          "toStateData": "${ .itemsToBuyAtStore }"
```

Comment:
idk what path then is. ".itemsTobuyAtStore" selects that element from the state data

Reply:
Sorry if my point is not very clear.  My point is that `"${ .itemsToBuyAtStore }"` reads like it is extracting some value from the state data, because the `${...}` implies that some expression is to be evaluated here.  Instead, I think this should be the string `".itemsToBuyAtStore"`(no `${}`) indicating that this is explicitly not an expression, that we're specifying a path in the data where the results should be inserted.

Reply:
oh..let's make that a different discussion as I feel that would require us to look at what other places that would / should be applied. is that ok ? for now all expressions used should be in ${ ... } format.

Reply:
Expressions vs. paths: #285